### PR TITLE
Add release / distribute support.

### PR DESCRIPTION
**Pull request by efeg**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -2,6 +2,43 @@
...
+      developer {
+        name "Adem Efe Gencer"
+        email "efegencer@gmail.com"
+      }
+      developer {
+        name "Jiangjie (Becket) Qin"
+        email "becket.qin@gmail.com"
+      }
+      developer {
+        name "Sir Joel Koshy"
```
Comment (radai-rosenblatt) [inline build.gradle]: :+1:

File: build.gradle [hunk 2]
```diff
@@ -2,6 +2,43 @@
...
+    scm {
+      url "https://github.com/linkedin/cruise-control"
+    }
+  }
+}
+
+
 allprojects {
   apply plugin: 'idea'
   apply plugin: 'eclipse'
```
Comment (radai-rosenblatt) [inline build.gradle]: does anyone really need the eclipse plugin ?!

File: build.gradle [hunk 3]
```diff
@@ -2,6 +2,43 @@
...
+    }
+    scm {
+      url "https://github.com/linkedin/cruise-control"
+    }
+  }
+}
+
+
 allprojects {
   apply plugin: 'idea'
```
Comment (radai-rosenblatt) [inline build.gradle]: vanilla gradle projects should not require the idea plugin

File: build.gradle [hunk 4]
```diff
@@ -62,6 +94,20 @@ subprojects {
 }
 
 project(':cruise-control-core') {
+  apply plugin: 'maven-publish'
+  apply plugin: 'com.jfrog.artifactory'
+
+  test.dependsOn('checkstyleMain', 'checkstyleTest')
```
Comment (radai-rosenblatt) [inline build.gradle]: can ,aybe move to the subprokect closure for java type ?

File: build.gradle [hunk 5]
```diff
@@ -93,6 +158,20 @@ project(':cruise-control-core') {
 
 project(':cruise-control') {
   apply plugin: 'scala'
+  apply plugin: 'maven-publish'
+  apply plugin: 'com.jfrog.artifactory'
+
+  test.dependsOn('checkstyleMain', 'checkstyleTest')
+
+  task sourcesJar(type: Jar, dependsOn: classes) {
```
Comment (radai-rosenblatt) [inline build.gradle]: sourcesJar and javadocJar may be easy to move to subProjects{} closure - they are identical always

File: build.gradle [hunk 6]
```diff
@@ -181,5 +293,53 @@ project(':cruise-control-metrics-reporter') {
...
+      publishArtifacts = true
+      publishPom = true
+      publishIvy = true
+    }
+  }
+  clientConfig.setIncludeEnvVars(false)
+}
+
+task distributeBuild(type: com.linkedin.gradle.build.DistributeTask) {
+//task distributeBuild(type: java.lang.Integer) {
```
Comment (radai-rosenblatt) [inline build.gradle]: commented out line

File: build.gradle (hunk 2 above)
Comment (efeg) [reply build.gradle]: Removed the plugin.

**Pull request merged**